src/main/java/uk/gov/pay/api/model/PaymentState.java

```diff
@@ -93,4 +95,24 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(status, finished, message, code);
     }
+
+    /* No-op setters are required to generate valid swagger.json file. Without the setters, attributes are generated as readonly and mandatory (which is invalid) in swagger file.
+       https://github.com/kongchen/swagger-maven-plugin/issues/458.
+       
+       @JsonProperty removed from variables as Jackson uses setters available to construct object. 
+       Instead constructor has been marked with @JsonCreator and @JsonProperty annotations and getters with @JsonGetter. 
```

Comment:
what's the benefit of using `@JsonGetter` over `@JsonCreator`? i'm dubious about changing this as it's different from the way we annotation other classes, e.g. uk.gov.pay.api.model.[Payment|PaymentEvent] etc.
also i don't like the no-op setters.

Reply:
We are using both @JsonGetter and @JsonCreator for this change. It is because of the issue with swagger plugin. Swagger generates invalid swagger.json if no-op setters are not available (generates fields as read only which is not the case). 

And if no-op setters are introduced, Jackson doesn't like the way we usually use @JsonProperty.

Reply:
Removed 'readOnly' for attributes. It is now inline with the rest of our code.